### PR TITLE
Update `default.qutrit` for new return types

### PR DESCRIPTION
**Pull request by mudit2812**
**Context:**
Updating `default.qutrit` to work with the new return types, and validating return types correctly.

**Description of the Change:**
* Updated `QutritDevice` to raise an error if `classical_shadow` or `shadow_expval` are measured on a qutrit device.
* Added tests to validate that `DefaultQutrit` works with the new return type system.

### CODE REVIEW

**Comment by mudit2812** [comment]
[sc-35002]

**Review by rmoyard — COMMENTED** [review]
It looks good to me ! Thank you for adding all the tests :100: Just a couple of suggestions and then I will approve it.

File: tests/returntypes/core/test_new_return_types_qnode.py [hunk 1]
```diff
@@ -175,23 +191,26 @@ def circuit(x):
 
     herm = np.diag([1, 2, 3, 4])
     probs_data = [
+        (qml.GellMann(0, 3), None),
+        (qml.THermitian(np.eye(9), wires=[1, 0]), None),
         (None, [0]),
         (None, [0, 1]),
         (qml.PauliZ(0), None),
         (qml.Hermitian(herm, wires=[1, 0]), None),
     ]
```
Comment (albi3ro) [inline tests/returntypes/core/test_new_return_types_qnode.py:200]: Might be cleaner to just have a `qubit_probs_data` and a `qutrit_probs_data`.

File: tests/returntypes/core/test_new_return_types_qnode.py [hunk 2]
```diff
@@ -988,14 +1067,16 @@ def circuit(x):
...
 
     @pytest.mark.parametrize("device", devices)
-    @pytest.mark.parametrize("op1,wires1,op2,wires2", multi_probs_data)
+    @pytest.mark.parametrize("op1,wires1,op2,wires2", multi_probs_data[4:])
```
Comment (albi3ro) [inline tests/returntypes/core/test_new_return_types_qnode.py:1070]: Same as above, might be better to just have two lists.

Comment (albi3ro) [reply tests/returntypes/core/test_new_return_types_qnode.py:1070]: We could also have an additional `qubit_devices` lists to parameterize over for tests like these.

**Review by albi3ro — APPROVED** [review]
Left a couple optional comments to improve the tests, but nothing that blocks the PR. Nice to see that it was just testing 🎉

**Review by rmoyard — APPROVED** [review]
Great job @mudit2812 :100:

**Pull request merged**